server/server.js

```diff
@@ -453,20 +447,80 @@ app.get("/user/getemployee", function(req, res) {
         console.log(err);
         res.send("Error getting employee from database.");
       } else {
-        console.log("Server 1100: " + rows[0].firstName);
-        var data = rows.map(x => ({
-          id: x.id,
-          firstName: x.firstName,
-          lastName: x.lastName,
-          email: x.email
-        }));
-        //console.log("Server: " + data[0].firstName + " " + data[0].id);
+        var data = rows.map((x) => ({ id: x.id, firstName: x.firstName, lastName: x.lastName, email: x.email }))
+        res.json(data);
+      }
+    }
+  );
+});
+
+app.get("/user/account", function(req, res) {
+  conn.query(
+  "SELECT id, firstName, lastName, email, password FROM user WHERE id=?",
+  [req.query.id],
+    function(err, rows) {
+      if (err) {
+        console.log(err);
+        res.send("Error getting employee from database.");
+      } else {
+        var data = rows.map((x) => ({ id: x.id, firstName: x.firstName, lastName: x.lastName, email: x.email, password: x.password }))
         res.json(data);
       }
     }
   );
 });
 
+app.post("/user/account", function(req, res) {
+  var changes = {
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    email: req.body.email
+  };
+  if (req.body.password != "") {
+    changes.password = req.body.password;
```

Comment:
The only issue I see is here we need to add the bcrypt stuff since we're not storing the password as plaintext anymore, but as a salted hash. See the /editUser or /addUser routes for how to implement.

Reply:
good catch! i made the changes already, and updated the pull request.